editor/code_editor.cpp

```diff
@@ -419,6 +419,26 @@ void FindReplaceBar::_update_matches_label() {
 	}
 }
 
+void FindReplaceBar::_temp_unfold_line(int p_line) {
+	// Unfold p_line, then walk back to find which was the first line that had to be unfolded so p_line can be shown.
+	// Mark that line as temporarily unfolded, so it's folded back on the next temporarily unfolding - unless the user for example edits the text.
+	if (temp_unfolded_line != -1) {
+		text_editor->fold_line(temp_unfolded_line);
+	}
+
+	if (!text_editor->is_line_folded(p_line)) {
+		temp_unfolded_line = p_line;
+		while (temp_unfolded_line > 0) {
+			if (text_editor->is_line_folded(temp_unfolded_line)) {
+				break;
+			}
+			temp_unfolded_line--;
+		}
```

Comment:
```suggestion
		do {
			if (text_editor->is_line_folded(temp_unfolded_line)) {
				break;
			}
			temp_unfolded_line--;
		} while (temp_unfolded_line >= 0);
```
This will ensure that if it finds no folded lines `temp_unfolded_line` will be `-1` and therefore not fold

Reply:
To clarify approval is contingent on this being fixed

Reply:
```suggestion
		while (temp_unfolded_line >= 0) {
			if (text_editor->is_line_folded(temp_unfolded_line)) {
				break;
			}
			temp_unfolded_line--;
		}
```
Should have the same effect, not treating the first line differently from the rest